dig_test.go

```diff
@@ -919,6 +919,48 @@ func TestGroups(t *testing.T) {
 		}), "invoke failed")
 	})
 
+	t.Run("groups are provided via option", func(t *testing.T) {
+		c := New(setRand(rand.New(rand.NewSource(0))))
+
+		provide := func(i int) {
+			require.NoError(t, c.Provide(func() int {
+				return i
+			}, Group("val")), "failed to provide ")
+		}
+
+		provide(1)
+		provide(2)
+		provide(3)
+
+		type in struct {
+			In
+
+			Values []int `group:"val"`
+		}
+
+		require.NoError(t, c.Invoke(func(i in) {
+			assert.Equal(t, []int{2, 3, 1}, i.Values)
```

Comment:
can avoid need for setting the rand seed by using [`assert.ElementsMatch`](https://godoc.org/github.com/stretchr/testify/assert#ElementsMatch)